### DISCUSSION TITLE
Email Configuration and Google SMTP Relay

### DISCUSSION THREAD

**Discussion by wesshaw** [Budibase/budibase#5864] [Help]
I've had Google SMTP Relay set up through our Google Workspace account for about a year.  I was initially able to add multiple users to my Budibase system by sending invitation emails and everything was great.  Over the past couple of weeks we've added a few new employees and I'm trying to get them registered in our Budibase apps.  Unfortunately, I'm now getting errors trying to send emails.  I'm getting this message: "Server terminates connection. response=421 4.7.0 Try again later, closing connection. (EHLO) v22-20020a9d69d6000000b00605f2223ff1sm158755oto.14 - gsmtp: 421 4.7.0 Try again later, closing connection. (EHLO) v22-20020a9d69d6000000b00605f2223ff1sm158755oto.14 - gsmtp" I'm using smtp-relay.gmail.com, None/STARTTLS, Port 587, and not requiring sign-in.  I have the relay configured correctly on Google's side to whitelist the IP of my self-hosted installation and the sender that I'm using.  Anybody have any ideas?  I've tried all of the permutations of configuration options and I'm not getting anywhere.  

Side note - and possibly a more important discussion - I'd love to use SSL/TLS (might solve my problem!), but I get this message: "140375378717512:error:1408F10B:SSL routines:ssl3_get_record:wrong version number:../deps/openssl/openssl/ssl/record/ssl3_record.c:331:\n" when trying to configure Email using SSL/TLS.

**Comment by mike12345567** [comment]
Hi @wesshaw - I had a look around for that error code and it looks like it might be something to do with your EHLO configuration in the relay: https://serverfault.com/questions/929559/postfix-error-421-4-7-0-try-again-later-closing-connection-ehlo

I'm not sure how much with the original error, but it would definitely be worth opening the second point as an issue and hopefully we can replicate it and see where the SSL version is being set incorrectly.

**Comment by jacquez1181** [comment]
I am having the same issue with the Gmail SMTP relay.  Failed to save email settings, reason: 139860814826312:error:1408F10B:SSL routines:ssl3_get_record:wrong version number:../deps/openssl/openssl/ssl/record/ssl3_record.c:331: 

Anywhere I can get logs to share these with the community?

**Reply by jacquez1181** [reply]
So I was able to get it to work by using SSL (Port 465) and authenticating with the budibase IP address instead of SMTP authentication.

**Reply by estelle-d** [reply]
Hello, 
I still can't make it work ;(
HOST: smtp.gmail.com or smtp-relay.gmail.com
Security type: SSL/TLS
Port: 25, 465, or 587

I have enable IMAP on my gmail account

which budibase IP address did you used to make it work?
Thanks,
Estelle

**Reply by estelle-d** [reply]
J'ai l'impression qu'il faut un compte google workspace pour que ca marche, est-ce correcte?

**Reply by nul0m** [reply]
> So I was able to get it to work by using SSL (Port 465) and authenticating with the budibase IP address instead of SMTP authentication.

@jacquez1181 and which IP address did you use? or you have the app self-hosted?

**Reply by peterhuppertz** [reply]
Was trying different settings and can't get it work
- Tried both ports 465 and 587. The error messages are different but non of them worked
  - Port 465 `Server terminates connection. response=421 4.7.0`
  - Port 587 `139729771887432:error:1408F10B:SSL routines:ssl3_get_record:wrong version number`
- Also changed the settings on google's side (Allowed sender: any address, require authentication yes/no)

Anyone solved this?

**Comment by DraculaVladimir** [comment]
https://support.google.com/accounts/answer/6010255?sjid=7688117697761545803-NC

Google made it so less secure applicatiosn can no longer connect via username and password. 
But it is not secure for other applications to have your email's password saved.
